src/Umbraco.Web.Common/ApplicationBuilder/UmbracoPipelineFilter.cs

```diff
@@ -2,43 +2,112 @@
 
 namespace Umbraco.Cms.Web.Common.ApplicationBuilder;
 
-/// <summary>
-///     Used to modify the <see cref="IApplicationBuilder" /> pipeline before and after Umbraco registers it's core
-///     middlewares.
-/// </summary>
-/// <remarks>
-///     Mainly used for package developers.
-/// </remarks>
+/// <inheritdoc />
 public class UmbracoPipelineFilter : IUmbracoPipelineFilter
 {
+    /// <summary>
+    /// Initializes a new instance of the <see cref="UmbracoPipelineFilter" /> class.
+    /// </summary>
+    /// <param name="name">The name.</param>
     public UmbracoPipelineFilter(string name)
-        : this(name, null, null, null)
-    {
-    }
+        : this(name, null, null, null, null, null)
+    { }
 
+    /// <summary>
+    /// Initializes a new instance of the <see cref="UmbracoPipelineFilter" /> class.
+    /// </summary>
+    /// <param name="name">The name.</param>
+    /// <param name="prePipeline">The pre pipeline callback.</param>
+    /// <param name="postPipeline">The post pipeline callback.</param>
+    /// <param name="endpointCallback">The endpoint callback.</param>
     public UmbracoPipelineFilter(
         string name,
         Action<IApplicationBuilder>? prePipeline,
         Action<IApplicationBuilder>? postPipeline,
         Action<IApplicationBuilder>? endpointCallback)
+        : this(name, prePipeline, null, null, postPipeline, endpointCallback)
+    { }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="UmbracoPipelineFilter" /> class.
+    /// </summary>
+    /// <param name="name">The name.</param>
+    /// <param name="prePipeline">The pre pipeline callback.</param> 
+    /// <param name="preRouting">The pre routing callback.</param>
+    /// <param name="postRouting">The post routing callback.</param>
+    /// <param name="postPipeline">The post pipeline callback.</param>
+    /// <param name="endpoints">The endpoints callback.</param>
+    public UmbracoPipelineFilter(
+        string name,
+        Action<IApplicationBuilder>? prePipeline = null,
+        Action<IApplicationBuilder>? preRouting = null,
+        Action<IApplicationBuilder>? postRouting = null,
+        Action<IApplicationBuilder>? postPipeline = null,
+        Action<IApplicationBuilder>? endpoints = null)
     {
         Name = name ?? throw new ArgumentNullException(nameof(name));
         PrePipeline = prePipeline;
+        PreRouting = preRouting;
+        PostRouting = postRouting;
         PostPipeline = postPipeline;
-        Endpoints = endpointCallback;
+        Endpoints = endpoints;
     }
 
+    /// <summary>
+    /// Gets or sets the pre pipeline callback.
+    /// </summary>
+    /// <value>
+    /// The pre pipeline callback.
+    /// </value>
     public Action<IApplicationBuilder>? PrePipeline { get; set; }
```

Comment:
Another suggestion for a future major (since it would be a breaking change), is to change the property setters to `init`, so you can only set the callback when constructing a new instance:

```suggestion
    public Action<IApplicationBuilder>? PrePipeline { get; init; }
```

This does raise the question whether we still want the constructor with all the optional parameters (which is currently already flagged as having too many parameters) or going the complete opposite way and completely removing the property setters...

Any thoughts on which syntax is preferred based on the following example?
```c#
// Using parameters will more likely produce longer lines or weird parameter alignment (if you break it over multiple lines)
builder.Services.Configure<UmbracoPipelineOptions>(options => options.AddFilter(new UmbracoPipelineFilter("Cors", postRouting: app => app.UseCors())))

// Using property setter will produce slightly more verbose code, but with nicer formatting
builder.Services.Configure<UmbracoPipelineOptions>(options => options.AddFilter(new UmbracoPipelineFilter("Cors")
{
    PostRouting = app => app.UseCors()
}));

```

Reply:
I personally prefer the first one, which means it's probably the wrong choice and the latter is best 😁

Reply:
There's nothing wrong with either choice, so let's leave this for now and allow both 😄